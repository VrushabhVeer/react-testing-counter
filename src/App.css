.App {
  margin-top: 30px;
  text-align: center;
}

button {
  padding: 8px 15px;
  font-size: 16px;
  font-weight: bold;
  color: white;
  border: 1px solid rgb(91, 87, 87);
  border-radius: 8px;
  margin: 5px;
}

.inc {
  background-color: chartreuse;
}

.dec {
  background-color: tomato;
}