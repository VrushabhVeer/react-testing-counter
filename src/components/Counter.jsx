import React, { useState } from 'react'

const Counter = () => {
    const [count, setCount] = useState(0);

    return (
        <counter data-testid="customComponent" >
            <h1>Counter</h1>
            <h2>{count}</h2>
            <button data-testid="incBtn" className='inc' onClick={() => setCount(count + 1)}>Increment</button>
            <button data-testid="decBtn" disabled={count === 0} className='dec' onClick={() => setCount(count - 1)}>Decrement</button>
        </counter>
    )
}

export default Counter;