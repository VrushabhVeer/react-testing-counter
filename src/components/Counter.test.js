import Counter from "./Counter";
import App from "../App";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";

describe("Testing the Counter component", () => {
  it("Should be present in the DOM", () => {
    render(<App />);
    let counter = screen.getByTestId("customComponent");

    expect(counter).toBeInTheDocument();
  });

  it("Should be present in the DOM", () => {
    render(<Counter />);
    let incButton = screen.getByText("Increment");

    expect(incButton).toBeInTheDocument();
  });

  it("Should be present in the DOM", () => {
    render(<Counter />);
    let decButton = screen.getByText("Decrement");

    expect(decButton).toBeInTheDocument();
  });

  it("Should click button", () => {
    render(<Counter />);

    let incButton = screen.getByTestId("incBtn");
    userEvent.click(incButton);

    let decButton = screen.getByTestId("decBtn");
    userEvent.click(decButton);
  });

  it("Should call the given function", () => {
    const mockFunc = jest.fn();
    render(<Counter onclick={mockFunc}>Increment</Counter>);

    const incButton = screen.getByTestId("incBtn");
    userEvent.click(incButton);
  });

  it("Should call the given function", () => {
    const mockFunc = jest.fn();
    render(<Counter onclick={mockFunc}>Decrement</Counter>);

    const decButton = screen.getByTestId("decBtn");
    userEvent.click(decButton);
  });

  it("should match the snapshot", () => {
    const tree = renderer.create(<Counter />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
