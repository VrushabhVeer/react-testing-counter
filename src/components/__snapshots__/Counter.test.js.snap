// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Counter component should match the snapshot 1`] = `
<counter
  data-testid="customComponent"
>
  <h1>
    Counter
  </h1>
  <h2>
    0
  </h2>
  <button
    className="inc"
    data-testid="incBtn"
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    className="dec"
    data-testid="decBtn"
    disabled={true}
    onClick={[Function]}
  >
    Decrement
  </button>
</counter>
`;
